Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -1251,8 +1262,8 @@ private static void CleanupCheckers(string requestId)
             verificationResult.Errors = null;
             verificationResult.Outcome = VCGen.Outcome.Inconclusive;
           }
-          catch(AggregateException ae)
-          {
+          catch(AggregateException ae) {
+            ThrowIfCanceled(ae);
```

Comment:
Similar to my previous comment, though perhaps it makes sense to have a separate method since this logic needs to happen in multiple places.